src/operator/svm_output-inl.h

```diff
@@ -0,0 +1,231 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file svm_output-inl.h
+ * \brief
+ * \author Jonas Amaro
+*/
+#ifndef MXNET_OPERATOR_SVM_OUTPUT_INL_H_
+#define MXNET_OPERATOR_SVM_OUTPUT_INL_H_
+
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator.h>
+#include <cstring>
+#include <map>
+#include <string>
+#include <vector>
+#include <utility>
+#include "./operator_common.h"
+
+namespace mxnet {
+namespace op {
+
+namespace svm_enum {
+enum SVMOutputOpInputs {kData, kLabel};
+enum SVMOutputOpOutputs {kOut};
+enum SVMOutputNormType {kNull, kBatch, kValid};
+enum SVMOutputOpResource {kTempSpace};
+}  // namespace svm_enum
+
+
+struct SVMOutputParam : public dmlc::Parameter<SVMOutputParam> {
+  float margin;
+  float regularization_coefficient;
+  bool use_linear;
+  DMLC_DECLARE_PARAMETER(SVMOutputParam) {
+    DMLC_DECLARE_FIELD(margin).set_default(1.0f)
+    .describe("Scale the DType(param_.margin) for activation size");
+    DMLC_DECLARE_FIELD(regularization_coefficient).set_default(1.0f)
+    .describe("Scale the coefficient responsible for balacing coefficient size and error tradeoff");
+    DMLC_DECLARE_FIELD(use_linear).set_default(false)
+    .describe("If set true, uses L1-SVM objective function. Default uses L2-SVM objective");
+  };
+};
+
+template<typename xpu, typename DType>
+class SVMOutputOp : public Operator {
+ public:
+  explicit SVMOutputOp(SVMOutputParam param) : param_(param) {}
+
+  virtual void Forward(const OpContext &ctx,
+                       const std::vector<TBlob> &in_data,
+                       const std::vector<OpReqType> &req,
+                       const std::vector<TBlob> &out_data,
+                       const std::vector<TBlob> &aux_args) {
+    using namespace mshadow;
+    using namespace mshadow::expr;
+    CHECK_EQ(in_data.size(), 2) << "Expecting [data, label]";
+    CHECK_EQ(out_data.size(), 1) << "Expecting [output]";
+    CHECK_EQ(req.size(), 1) << "Expecting output.size() == req.size()";
+    Stream<xpu> *s = ctx.get_stream<xpu>();
+    Tensor<xpu, 2, DType> data = in_data[svm_enum::kData].FlatTo2D<xpu, DType>(s);
+    Tensor<xpu, 2, DType> out = out_data[svm_enum::kOut].FlatTo2D<xpu, DType>(s);
+    out = DType(0.0f);
+    out += data;
```

Comment:
Here we can use `Assign(out, req[svm_enum::kOut], F<mshadow_op::identity>(data));`